File: 프로그래머스/0/120880. 특이한 정렬/특이한 정렬.java
import java.util.*;
class Solution {
    public int[] solution(int[] num, int n) {
        int[] answer = {};
        int m = num.length;
        for(int i = 0; i < m; i++){
            for(int j = 0; j < m - 1; j++){
                if(Math.abs(num[j]-n) == Math.abs(num[j + 1] - n) && num[j] > num[j + 1]) continue;
                if(Math.abs(num[j] - n) >= Math.abs(num[j + 1] - n)){
                    int temp = num[j];
                    num[j] = num[j + 1];
                    num[j + 1] = temp;
                }
            }
        }
        return num;
    }
}

File: 프로그래머스/0/120880. 특이한 정렬/특이한 정렬.py


def solution(numlist, n):
    return sorted(numlist, key = lambda x: (abs(n - x), -x))

"""
1.n기준 가까운 순으로 정렬
2.거리 같으면 큰수가 앞으로 오도록 배치
3.
"""